geniza/corpus/views.py

```diff
@@ -646,6 +651,74 @@ def form_valid(self, form):
         return super(DocumentMerge, self).form_valid(form)
 
 
+class SourceAutocompleteView(PermissionRequiredMixin, autocomplete.Select2QuerySetView):
+    permission_required = ("corpus.change_document",)
+
+    def get_queryset(self):
+        """sources filtered by entered query, or all sources, ordered by author last name"""
+        q = self.request.GET.get("q", None)
+        qs = Source.objects.all().order_by("authors__last_name")
+        if q:
+            qs = qs.filter(
+                Q(title__icontains=q)
+                | Q(authors__first_name__istartswith=q)
+                | Q(authors__last_name__istartswith=q)
+                | Q(year__istartswith=q)
+                | Q(journal__icontains=q)
+                | Q(notes__icontains=q)
+                | Q(other_info__icontains=q)
+                | Q(languages__name__icontains=q)
+                | Q(volume__icontains=q)
+            ).distinct()
+        return qs
+
+
+class DocumentAddTranscriptionView(PermissionRequiredMixin, CreateView):
+    permission_required = ("corpus.change_document",)
+    template_name = "corpus/add_transcription_source.html"
+    viewname = "corpus:document-add-transcription"
+    model = Footnote
+    form_class = FootnoteInlineForm
+
+    def get_object(self):
+        """Get Document instead of Footnote"""
+        return Document.objects.get(pk=self.kwargs.get("pk"))
+
+    def page_title(self):
+        """Title of add transcription page"""
+        return "Add a new transcription for %(doc)s" % {"doc": self.get_object().title}
+
+    def post(self, request, *args, **kwargs):
+        """Create footnote linking source to document, then redirect to edit transcription view"""
+        source_pk = int(
+            request.POST["footnotes-footnote-content_type-object_id-0-source"]
+        )
+        source = Source.objects.get(pk=source_pk)
+        Footnote.objects.get_or_create(
+            source=source,
+            doc_relation=[Footnote.EDITION],
+            content_type=ContentType.objects.get_for_model(Document),
+            object_id=self.kwargs.get("pk"),
+        )
```

Comment:
We don't need to create the footnote here, or even retrieve the source — footnote will be created when/if annotations are added in the new signal handler you have in a different branch. Just redirect.